### PR TITLE
TLS and Basic Auth support

### PR DESCRIPTION
**Pull request by mahendrapaipuri**
This PR adds the TLS and basic auth support to the exporter based on [exporter toolkit](https://github.com/prometheus/exporter-toolkit/tree/master). TLS and basic auth are opt-in and disabled by default. Users can enable them by passing a [YAML file](https://github.com/prometheus/exporter-toolkit/blob/master/docs/web-config.yml) that has both TLS and basic auth config using `--web-config-file` flag.

This PR completes #122 which has been stale for a while.

Cheers!

### CODE REVIEW

**Comment by mahendrapaipuri** [comment]
Hello, any chance that maintainers are interested in this feature?

**Comment by glowkey** [comment]
Yes, thank you for the contribution. We plan on reviewing this in the near future and before the next major release.

**Comment by mahendrapaipuri** [comment]
@glowkey Cheers for response. That is awesome. TLS and auth would be very handy for our HPC platform.

File: cmd/dcgm-exporter/main.go [hunk 1]
```diff
@@ -174,6 +177,21 @@ func main() {
...
 		},
+		&cli.StringFlag{
+			Name:    CLIWebConfigFile,
+			Value:   "",
+			Usage:   "TLS config file following webConfig spec.",
+			EnvVars: []string{"DCGM_EXPORTER_WEB_CONFIG_FILE"},
+		},
+	}
+
+	if runtime.GOOS == "linux" {
```
[the remark this answers is not in the document]
Comment (mahendrapaipuri) [reply cmd/dcgm-exporter/main.go:188]: Added the if branch to exit if `GOOS` is not linux

[the remark this answers is not in the document]
Comment (nvvfedorov) [reply cmd/dcgm-exporter/main.go:188]: Thanks

**Review by glowkey — APPROVED** [review]

**Review by nvvfedorov — APPROVED** [review]
LGTM

**Pull request merged**